File: LISTAS/PROYECTO GENERAL PYTHON/MENU.py
import os
import subprocess
import msvcrt

def get_key():
    while True:
        if msvcrt.kbhit():
            key = msvcrt.getch()
            if key == b'\xe0':  # Special keys (arrows, f keys, ins, del, etc.)
                key = msvcrt.getch()
                return key
            else:
                return key

def print_menu(selected_index):
    menu_items = ["Lista Simple", "Lista Doble", "Lista Circular Doble", "Lista Circular Simple", "Salir"]
    print("========================================")
    print("      Menu de Proyectos de Listas")
    print("========================================")
    for i, item in enumerate(menu_items):
        if i == selected_index:
            print(f" > {item} <")
        else:
            print(f"   {item}")
    print("========================================")

def main():
    rutas = {
        0: {
            "ruta_autos": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\autos.txt",
            "ruta_autos_historial": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\autos_historial.txt",
            "ruta_propietarios": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\propietarios.txt",
            "ruta_exe": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\ListaSimple.exe",
            "ruta_dir": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE"
        },
        1: {
            "ruta_autos": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\autos.txt",
            "ruta_autos_historial": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\autos_historial.txt",
            "ruta_propietarios": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\propietarios.txt",
            "ruta_exe": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\ListaDoble.exe",
            "ruta_dir": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE"
        },
        2: {
            "ruta_autos": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\autos.txt",
            "ruta_autos_historial": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\autos_historial.txt",
            "ruta_propietarios": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\propietarios.txt",
            "ruta_exe": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\ListaCircularDoble.exe",
            "ruta_dir": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE"
        },
        3: {
            "ruta_autos": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR SIMPLE\\autos.txt",
            "ruta_autos_historial": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR SIMPLE\\autos_historial.txt",
            "ruta_propietarios": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR SIMPLE\\propietarios.txt",
            "ruta_exe": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR SIMPLE\\ListaCircularSimple.exe",
            "ruta_dir": "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR SIMPLE"
        }
    }

    selected_index = 0
    while True:
        os.system('cls')  # Use 'cls' for Windows
        print_menu(selected_index)
        key = get_key()
        if key == b'H':  # Up arrow
            selected_index = (selected_index - 1) % 5
        elif key == b'P':  # Down arrow
            selected_index = (selected_index + 1) % 5
        elif key == b'\r':  # Enter key
            if selected_index == 4:
                print("Saliendo del programa...")
                break
            else:
                ruta = rutas.get(selected_index)
                if ruta:
                    os.chdir(ruta["ruta_dir"])
                    subprocess.run(ruta["ruta_exe"], shell=True)
                else:
                    print("Opción no válida.")
                break

if __name__ == "__main__":
    main()

File: LISTAS/PROYECTO GENERAL/Menu.cpp
#include <iostream>
#include <conio.h>
#include <windows.h>  // For system("cls") and SetCurrentDirectory
#include <vector>
#include <string>

using namespace std;

int mostrarMenu(const vector<string>& opciones) {
    int seleccion = 0;
    int numOpciones = opciones.size();

    while (true) {
        system("cls");  // Clear the screen

        cout << "========================================" << endl;
        cout << "      Menu de Proyectos de Listas       " << endl;
        cout << "========================================" << endl;

        for (int i = 0; i < numOpciones; ++i) {
            if (i == seleccion) {
                cout << " > " << opciones[i] << " < " << endl;  
            } else {
                cout << "   " << opciones[i] << endl; 
            }
        }

        cout << "========================================" << endl;

      
        char tecla = _getch();

        
        if (tecla == 'w' || tecla == 72) {
            seleccion = (seleccion - 1 + numOpciones) % numOpciones;
        } else if (tecla == 's' || tecla == 80) {
            seleccion = (seleccion + 1) % numOpciones;
        } else if (tecla == 13) {  
            return seleccion;
        }
    }
}

int main() {
    string rutaAutos, rutaAutosHistorial, rutaPropietarios;
    string rutaExe;
    string rutaDir;

    vector<string> opciones = {"Lista Simple", "Lista Doble", "Lista Circular Doble", "Lista Circular Simple", "Salir"};
    int seleccion = mostrarMenu(opciones);

    switch (seleccion) {
        case 0: 
            rutaAutos = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\autos.txt";
            rutaAutosHistorial = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\autos_historial.txt";
            rutaPropietarios = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\propietarios.txt";
            rutaExe = "\"C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE\\ListaSimple.exe\"";
            rutaDir = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE";
            break;
        case 1: 
            rutaAutos = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\autos.txt";
            rutaAutosHistorial = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\autos_historial.txt";
            rutaPropietarios = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\propietarios.txt";
            rutaExe = "\"C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE\\ListaDoble.exe\"";
            rutaDir = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA DOBLE";
            break;
        case 2: 
            rutaAutos = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\autos.txt";
            rutaAutosHistorial = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\autos_historial.txt";
            rutaPropietarios = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\propietarios.txt";
            rutaExe = "\"C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE\\ListaCircularDoble.exe\"";
            rutaDir = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR DOBLE";
            break;
        case 3: 
            rutaAutos = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA CIRCULAR SIMPLE\\autos.txt";
            rutaAutosHistorial = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE CIRCULAR\\autos_historial.txt";
            rutaPropietarios = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE CIRCULAR\\propietarios.txt";
            rutaExe = "\"C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE CIRCULAR\\ListaCircularSimple.exe\"";
            rutaDir = "C:\\REPOSITORIO\\PROYECTO-ESTRCUTURA-1978\\LISTAS\\LISTA SIMPLE CIRCULAR";
            break;
        case 4:
            cout << "Saliendo del programa..." << endl;
            return 0;
        default:
            cout << "Opción no válida." << endl;
            return 1;
    }

    // Cambiar el directorio de trabajo
    if (!SetCurrentDirectory(rutaDir.c_str())) {
        cerr << "Error al cambiar el directorio de trabajo a " << rutaDir << endl;
        return 1;
    }

    // Ejecutar el archivo .exe correspondiente
    system(rutaExe.c_str());

    return 0;
}